stock_account_inventory_force_date/models/stock_inventory.py

```diff
@@ -49,12 +49,25 @@ def _generate_lines_at_date(self, location, product_ids):
         }).search([('id', 'in', product_ids)]).filtered(
             lambda p: p.qty_available)
         for p in products_at_date:
-            line = {
-                'product_id': p.id,
-                'product_qty': p.qty_available,
-                'location_id': location.id,
-            }
-            lines.append(line)
+            if p.tracking == "none":
+                line = {
+                    'product_id': p.id,
+                    'product_qty': p.qty_available,
+                    'location_id': location.id,
+                }
+                lines.append(line)
+            else:
+                res = self.env["stock.move.line"].get_lot_qty_at_date_in_location(
```

Comment:
Should we put a `lot.ensure_one()` or an assert to ensure that `lot` has been sent as parameter if the product is tracked by lot?

Reply:
No, because this function must work properly with or without a lot.
If I have a product tracked by lot and I want to inventory it, without specifying a lot, we will have `None` in lot parameter.
But if I want to inventory only a specific lot, the parameter will restrict results to this specific lot.